### PR TITLE
Support creating subscription products (simple/variable) from scratch

### LINKED ISSUE #10173: Logic to add subscription default values when creating product from scratch
Use the default values set in `wp-admin` when creating a subscription product from scratch

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #10173 

NOTE: Don't panic about the number of file changes, most of them are string.xml removals. 

### Description

Enables creation subscription products (either simple or variable) from scratch in the app. 

### Testing instructions

1. Navigate to the product list and click + product button. 
2. Select simple subscription product type.
3. Add any details to the product, specifically all the subscription detail you can add. 
4. Save or publish the product. 
5. Check all the subscription data was properly sync with the API by checking the new product in `wp-admin`

🔁 The steps for a variation subscription product. 

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/2663464/77fa6fcd-6e02-4a71-bd0c-3bb82eb5c833

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductNavigationTarget.kt [hunk 1]
```diff
@@ -36,7 +36,6 @@ sealed class ProductNavigationTarget : Event() {
     data class ViewProductVariations(
         val remoteId: Long,
         val productSelectorFlow: ProductSelectorFlow = ProductSelectorFlow.Undefined,
-        val isReadOnlyMode: Boolean = false
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductNavigationTarget.kt:39]: This field was only used for subscription variations. Now that these are editable I've got rid of it

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/domain/GenerateVariationCandidates.kt [hunk 2]
```diff
@@ -21,7 +21,7 @@ class GenerateVariationCandidates @Inject constructor(
     }
 
     suspend operator fun invoke(product: Product): List<VariationCandidate> {
-        if (product.type != ProductType.VARIABLE.value) {
+        if (product.type != ProductType.VARIABLE.value && product.type != ProductType.VARIABLE_SUBSCRIPTION.value) {
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/domain/GenerateVariationCandidates.kt]: np, we can probably simplify this using the function you added `isVariableProduct` 🤔

**Review by hichamboushaba — APPROVED** [review]
Nice job, works well, thanks @JorgeMucientes

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/domain/GenerateVariationCandidates.kt (hunk 2 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/domain/GenerateVariationCandidates.kt]: Good catch! Added

**Pull request merged**